src/visual.ts

```diff
@@ -1900,6 +1902,55 @@ export class MekkoChart implements IVisual {
         };
     }
 
+    private applyGridSettings(): void {
+        const gridStyle = this.settingsModel.valueAxis.gridStyle?.value;
+        const gridWidth = this.settingsModel.valueAxis.gridWidth?.value;
+        const gridScale = this.settingsModel.valueAxis.gridScale?.value;
+        let dashArray = "none";
+        let lineCap = this.settingsModel.valueAxis.gridDashCap?.value;
+
+        switch (gridStyle) {
+            case "dashed":
+                dashArray = `${gridWidth * 4}, ${gridWidth * 2}`;
+                break;
+            case "dotted":
+                dashArray = `${gridWidth * 0.1}, ${gridWidth * 3}`;
+                lineCap = "round";
+                break;
+            case "custom":
+                const customPattern = this.settingsModel.valueAxis.gridDashArray?.value;
+                dashArray = customPattern;
+                // Scale the dash pattern by gridWidth if it's a numeric pattern
+                if (gridScale) {
+                    dashArray = dashArray
+                        .split(",")
+                        .map(s => parseFloat(s.trim()) * gridWidth)
+                        .join(", ");
+                }
+                break;
+            case "solid":
+            default:
+                dashArray = "none";
+                break;
+        }
+
+        this.y1AxisGraphicsContext
+            .selectAll(".tick line")
+            .style("stroke", this.settingsModel.valueAxis.gridColor.value.value)
+            .style("stroke-width", this.settingsModel.valueAxis.gridWidth.value)
+            .style("stroke-dasharray", dashArray)
+            .style("stroke-linecap", lineCap)
+            .style("opacity", (100 - this.settingsModel.valueAxis.gridTransparency.value) / 100);
+
+        // // Customize the main axis line (domain line)
+        this.y1AxisGraphicsContext
+            .select(".domain")
+            .style("stroke-width", "0px");
+        this.y1AxisGraphicsContext
+            .select(".tick line:last-of-type")
+            .style("stroke-width", 0);
```

Comment:
better to keep similar code style
`.style("stroke-width", "0px");` also defining "0px" is better for CSS, even if 0 is also good for d3